plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionData.java

```diff
@@ -166,6 +166,8 @@ public static Object getValue(JsonNode partitionValue, Type type)
                 return rescale(
                         partitionValue.decimalValue(),
                         createDecimalType(decimalType.precision(), decimalType.scale()));
+            // TODO https://github.com/trinodb/trino/issues/19753 Support Iceberg timestamp types with nanosecond precision
+            case TIMESTAMP_NANO:
```

Comment:
FYI: there's also `io.trino.plugin.iceberg.catalog.glue.GlueIcebergUtil#toGlueTypeStringLossy`. It has a `default` clause, which makes the `switch` not subject to exhaustivity checks.